client/modules/_hooks/src/datafiles/useRename.ts

```diff
@@ -0,0 +1,25 @@
+import { useMutation } from '@tanstack/react-query';
+import apiClient from '../apiClient';
+
+type TRenameParam = { api: string; system: string; path: string; name: string };
+
+function renameFn(src: TRenameParam, dest: TRenameParam, doi?: string) {
```

Comment:
`src` and `dest` make sense as arguments for move/copy since the system and path could change. For rename, I think there should just be a single param object that has the API, system, path, and new name.